### PR TITLE
test: fix url checks in e2e tests

### PR DESCRIPTION
**Pull request by aslilac**
Prevents flakes on pages like `WorkspacePage`, where query params are added after it has loaded. If they get added between polls, `toHaveURL` will fail, because it matches the full URL.

### CODE REVIEW

**Review by Emyrk — APPROVED** [review]
The `toHaveURL` takes a string or a regex. I would guess the regex option would pass, and I guess the string option does not act like a regex match?? Kinda ridiculous imo the behavior is so different based on if you passed in a regex string or just a string.

Would be nice if we could add a linter or remove the `toHaveURL` somehow to prevent misuse.

File: site/src/pages/CreateTokenPage/CreateTokenForm.tsx [hunk 1]
```diff
@@ -116,6 +116,7 @@ export const CreateTokenForm: FC<CreateTokenFormProps> = ({
 
             {lifetimeDays === "custom" && (
               <TextField
+                data-chromatic="ignore"
```
Comment (Emyrk) [inline site/src/pages/CreateTokenPage/CreateTokenForm.tsx:119]: Not sure what this is :thinking:

Comment (aslilac) [reply site/src/pages/CreateTokenPage/CreateTokenForm.tsx:119]: someone let a leaky date sneak in to chromatic again, so I'm muting it 😅

File: site/e2e/expectUrl.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { expect, type Page } from "@playwright/test";
+
+type PollingOptions = { timeout?: number; intervals?: number[] };
+
+export const expectUrl = expect.extend({
```
[the remark this answers is not in the document]
Comment (aslilac) [reply site/e2e/expectUrl.ts:5]: I think having several of them would be a fine outcome personally. This sort of "scoped custom `expect`s" is what the Playwright documentation suggests.

**Comment by aslilac** [comment]
> Would be nice if we could add a linter or remove the toHaveURL somehow to prevent misuse.

yeah, in the mean time I'm hoping people will mostly copy-paste from other tests that do the correct thing. :p

**Pull request merged**